packages/framework/src/types/schema.types/zod.schema.types.ts

```diff
@@ -0,0 +1,42 @@
+import zod from 'zod';
+
+export type ZodSchema = zod.ZodType;
+
+/**
+ * A minimal ZodSchema type.
+ *
+ * It is necessary to define a minimal ZodSchema type to enable correct inference
+ * when Zod is not available, as Typescript doesn't support detection of module
+ * availability via `typeof import('zod')`.
+ */
+export type ZodSchemaMinimal = {
+  readonly safeParseAsync: unknown;
+};
```

Comment:
This is the new minimal zod schema for use when Zod is not available in `node_modules`. It ensures we can defer to a `Record<string, unknown>` type after exhausting all schema possibilities.